### PR TITLE
CO-3830 : change of email trigger a merge of mass.mailing.contact

### PR DESCRIPTION
**Pull request by TheRaphael0000**
the remove condition was not correct and it should only remove the contract when these is no more partner in the contract

### CODE REVIEW

File: mass_mailing_switzerland/models/res_partner.py [hunk 1]
```diff
@@ -105,12 +105,15 @@ def write(self, vals):
                 mailing_contact_vals["opt_out"] = vals["opt_out"]
             if "category_id" in vals:
                 mailing_contact_vals["tag_ids"] = vals["category_id"]
-            if "email" in vals and not vals["email"] and not self.env.context.get(
+            if "email" in vals and vals["email"] and not self.env.context.get(
```
Comment (ecino) [inline mass_mailing_switzerland/models/res_partner.py]: I think we want to include the case were we remove the email address
```suggestion
            if "email" in vals and not self.env.context.get(
```

File: mass_mailing_switzerland/models/res_partner.py [hunk 2]
```diff
@@ -105,12 +105,15 @@ def write(self, vals):
...
                     "import_from_mailchimp"):
+                mailing_contacts.write({"partner_ids": [(3, self.id)]})
                 # In this case we don't need anymore the mailchimp contacts
-                mailing_contacts.unlink()
+                if len(mailing_contacts.partner_ids) <= 0:
+                    mailing_contacts.unlink()
```
Comment (ecino) [inline mass_mailing_switzerland/models/res_partner.py:116]: I think we miss a part to sync the new email to a mass mailing contact, no? Please add that.

**Review by ecino — CHANGES_REQUESTED** [review]
I think the fix is wrong. The logic before the fix was to check if we remove the email from a partner, then we delete the mass mailing contacts associated.

Now you changed this to: if we change the email of a partner, the mailing contacts are removed. But I think that the new email address won't be exported to mailchimp!

File: mass_mailing_switzerland/models/res_partner.py (hunk 2 above)
Comment (TheRaphael0000) [reply mass_mailing_switzerland/models/res_partner.py:116]: I have add the code for this below (I dont understand why it won't ask a new review)

**Review by ecino — APPROVED** [review]

**Pull request merged**